src/Workspaces/Remote/ServiceHub/Services/CodeAnalysis/CodeAnalysisService_SemanticClassificationCache.cs

```diff
@@ -143,60 +148,71 @@ private static void WriteTo(List<ClassifiedSpan> classifiedSpans, ObjectWriter w
             foreach (var type in classificationTypes)
                 writer.WriteString(type);
 
-            // Now emit each classified span as a triple of it's type, start, length.
+            // Now emit each classified span as a triple of it's start, length, type.
+            //
+            // In general, the latter two will all be a single byte as tokens tend to be short and we don't have many
+            // classification types.
+            //
+            // We do need to store the start (as opposed to a delta) as we may have multiple items starting at teh same
+            // position and we cannot encode a negative delta.
             writer.WriteInt32(classifiedSpans.Count);
             foreach (var classifiedSpan in classifiedSpans)
             {
-                writer.WriteInt32(seenClassificationTypes[classifiedSpan.ClassificationType]);
-                writer.WriteInt32(classifiedSpan.TextSpan.Start);
-                writer.WriteInt32(classifiedSpan.TextSpan.Length);
+                checked
+                {
```

Comment:
Any reason not to add WriteCompressedInt() helpers to avoid the explicit checked blocks? (You might have to move them into the helper but at least then it's less scary at use.)

Reply:
primarily because i didn't want to have to think about the encoding strategy used orecorner cases that the core lib might have to care about.  here, i only had to think about the set of values of ints that could fit into a uint.  Those are values up to 2B (technically 2<<30 given what ObjectWriter supports), which far exceeds any values that we coudl ever reasonably expect to get during classification for length.